components/datadog/apps/tracegen/k8s.go

```diff
@@ -0,0 +1,197 @@
+			Template: &corev1.PodTemplateSpecArgs{
+				Metadata: &metav1.ObjectMetaArgs{
+					Labels: pulumi.StringMap{
+						"app": pulumi.String("tracegen-uds"),
+					},
+				},
+				Spec: &corev1.PodSpecArgs{
+					Affinity: affinity,
+					Containers: corev1.ContainerArray{
+						&corev1.ContainerArgs{
+							Name:  pulumi.String("tracegen"),
+							Image: pulumi.String("ghcr.io/datadog/apps-tracegen:main"),
+							Env: &corev1.EnvVarArray{
+								&corev1.EnvVarArgs{
+									Name:  pulumi.String("DD_TRACE_AGENT_URL"),
+									Value: pulumi.String(traceAgentURL),
+								},
+							},
+							Resources: &corev1.ResourceRequirementsArgs{
+								Limits: pulumi.StringMap{
+									"cpu":    pulumi.String("100m"),
+									"memory": pulumi.String("32Mi"),
+								},
+								Requests: pulumi.StringMap{
+									"cpu":    pulumi.String("10m"),
+									"memory": pulumi.String("32Mi"),
+								},
+							},
+							VolumeMounts: &corev1.VolumeMountArray{
+								&corev1.VolumeMountArgs{
+									Name:      pulumi.String("apmsocketpath"),
+									MountPath: pulumi.String("/var/run/datadog"),
+								},
+							},
+						},
+					},
+					Volumes: corev1.VolumeArray{
+						&corev1.VolumeArgs{
+							Name: pulumi.String("apmsocketpath"),
+							HostPath: &corev1.HostPathVolumeSourceArgs{
+								Path: pulumi.String("/var/run/datadog"),
+							},
+						},
+					},
+				},
+			},
+		},
+	}, opts...); err != nil {
+		return nil, err
+	}
+
+	if _, err := appsv1.NewDeployment(e.Ctx, fmt.Sprintf("%s/tracegen-udp", namespace), &appsv1.DeploymentArgs{
+		Metadata: &metav1.ObjectMetaArgs{
+			Name:      pulumi.String("tracegen-udp"),
+			Namespace: pulumi.String(namespace),
+			Labels: pulumi.StringMap{
+				"app": pulumi.String("tracegen-udp"),
+			},
+		},
+		Spec: &appsv1.DeploymentSpecArgs{
+			Replicas: pulumi.Int(1),
+			Selector: &metav1.LabelSelectorArgs{
+				MatchLabels: pulumi.StringMap{
+					"app": pulumi.String("tracegen-udp"),
+				},
+			},
+			Template: &corev1.PodTemplateSpecArgs{
+				Metadata: &metav1.ObjectMetaArgs{
+					Labels: pulumi.StringMap{
+						"app": pulumi.String("tracegen-udp"),
+					},
+				},
+				Spec: &corev1.PodSpecArgs{
+					Affinity: affinity,
+					Containers: corev1.ContainerArray{
+						&corev1.ContainerArgs{
+							Name:  pulumi.String("tracegen-udp"),
+							Image: pulumi.String("ghcr.io/datadog/apps-tracegen:main"),
+							Env: &corev1.EnvVarArray{
+								&corev1.EnvVarArgs{
+									Name: pulumi.String("DD_AGENT_HOST"),
+									ValueFrom: &corev1.EnvVarSourceArgs{
+										FieldRef: &corev1.ObjectFieldSelectorArgs{
+											FieldPath: pulumi.String("status.hostIP"),
+										},
+									},
+								},
+								&corev1.EnvVarArgs{
+									Name: pulumi.String("DD_ENTITY_ID"),
+									ValueFrom: &corev1.EnvVarSourceArgs{
+										FieldRef: &corev1.ObjectFieldSelectorArgs{
+											FieldPath: pulumi.String("metadata.uid"),
+										},
+									},
+								},
```

Comment:
✂️ This is irrelevant for the tracer.

Reply:
Removed !
Also I think it could be useful to keep it as the admission controller adds it and we should check that the expected behavior isn't affected.